camel/embeddings/openai_embedding.py

```diff
@@ -56,7 +57,11 @@ def __init__(
             assert isinstance(dimensions, int)
             self.output_dim = dimensions
         self._api_key = api_key or os.environ.get("OPENAI_API_KEY")
-        self.client = OpenAI(timeout=60, max_retries=3, api_key=self._api_key)
+        self._url = url or os.environ.get("OPENAI_API_BASE_URL")
+        client_kwargs = {"timeout": 60, "max_retries": 3, "api_key": self._api_key}
+		if self._url:
+			client_kwargs["base_url"] = self._url
+		self.client = OpenAI(**client_kwargs)
```

Comment:
Thanks @Tsumugii24 ! The code here can be more unified with other model client setting part of camel
```
        self._client = OpenAI(
            timeout=60,
            max_retries=3,
            base_url=self._url,
            api_key=self._api_key,
        )
```